kernelci/api/helper.py

```diff
@@ -217,6 +217,47 @@ def submit_results(self, results, root):
         except requests.exceptions.HTTPError as error:
             raise RuntimeError(json.loads(error.response.text)) from error
 
+    def get_node_obj(self, node_dict, get_linked=False):
+        """Takes a dict defining a Node and returns it as a concrete
+        Node object (or Node subtype object). If get_linked is set to
+        True, linked nodes are vivified (not recursively).
+
+        It will also accept a Node (or subclass) object instead of a
+        dict. This can be used to fetch and vivify the linked objects if
+        get_linked is set to True.
+        """
+        def get_attr(obj, attr):
+            """Similar behavior to the builtin getattr, but it can be
+            used with nested attributes in.dot.notation
+            """
+            fields = attr.split('.')
+            if len(fields) == 1:
+                return getattr(obj, fields[0])
+            return get_attr(getattr(obj, fields[0]), '.'.join(fields[1:]))
+
+        def set_attr(obj, attr, value):
+            """Similar behavior to the builtin setattr, but it can be
+            used with nested attributes in.dot.notation
+            """
+            fields = attr.split('.')
+            if len(fields) == 1:
+                setattr(obj, fields[0], value)
+                return
+            obj = get_attr(obj, '.'.join(fields[:-1]))
+            setattr(obj, fields[-1], value)
+
+        generic_node_obj = models.Node.parse_obj(node_dict)
+        node_obj = models.parse_node_obj(generic_node_obj)
+        if get_linked:
+            for linked_node_attr in node_obj._OBJECT_ID_FIELDS:
+                node_id = get_attr(node_obj, linked_node_attr)
+                if node_id:
+                    resp = self.api.node.get(node_id)
+                    linked_node_obj = models.Node.parse_obj(resp)
+                    linked_obj = models.parse_node_obj(linked_node_obj)
```

Comment:
Would it make sense to improve `models.parse_node_obj()` so it:
* detects if its argument is a dict
* parse it as a generic Node first
* then moves on trying to make it a subclass of Node